src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -100,7 +101,15 @@ public RestoreCommand()
 
                 Console.PrintPackageSources(packageSources);
 
-                var localCache = new NuGetv3LocalRepository(packagesDir);
+                var sharedCache = new RestoreCommandSharedCache();
+                sharedCache.LocalCache = new NuGetv3LocalRepository(packagesDir);
```

Comment:
should LocalCache be a property? I'm thinking about scenarios where someone else is using it, and forgets to set the local cache? make it a constructor argument?